cmd/create/cluster/cmd.go

```diff
@@ -1569,14 +1569,19 @@ func run(cmd *cobra.Command, _ []string) {
 
 		// Validate subnets in the case the user has provided them using the `args.subnets`
 		if subnetsProvided {
-			if isHostedCP {
-				// Hosted cluster should validate that
-				// - Public hosted clusters have at least one public subnet
-				// - Private hosted clusters have all subnets private
-				privateSubnetsCount, err = ocm.ValidateHostedClusterSubnets(awsClient, privateLink, subnetIDs)
-			} else {
-				err = ocm.ValidateSubnetsCount(multiAZ, privateLink, len(subnetIDs))
+			err = ocm.ValidateSubnetsCount(multiAZ, privateLink, len(subnetIDs))
```

Comment:
Does this check applies also for Hypershift clusters?

Reply:
it's included in https://github.com/openshift/rosa/blob/master/pkg/ocm/helpers.go#L641

Reply:
Just asking because previously we didn't run this check for Hypershift clusters:
![image](https://user-images.githubusercontent.com/57869309/226902364-896ee6d7-955a-4eb4-bb08-0a1509cd1b7c.png)
Let me know if I missed anything.

Reply:
In both interactive/non-interactive that validation is perform at this point:

https://github.com/openshift/rosa/pull/1123/files#diff-943c0ab57b82c7028652df55e1d49e4f4e4dc41fa094dc4a9b3f2c35887d44b5R1580

Reply:
No. Line 1572 is when subnets are provided for classic clusters.

For interactive and subnets provider, the hosted cluster needs to evaluate the "ValidateHostedClusterSubnets" that's what's is out of that conditional block.

Reply:
Ouch, now I see it :).
Good catch, yes, that's missing.

Reply:
Added.